subprojects/docs/src/docs/userguide/commandLineInterface.adoc

```diff
@@ -0,0 +1,520 @@
+
+++++
+<figure>
+    <title>Example Task Graph</title>
+    <imageobject>
+        <imagedata fileref="img/commandLineTutorialTasks.png"/>
+    </imageobject>
+</figure>
+++++
+
+++++
+<sample id="excludeTask" dir="userguide/tutorial/excludeTasks" title="Excluding tasks">
+    <output args="dist --exclude-task test"/>
+</sample>
+++++
+
+You can see that the `test` task is not executed, even though it is a dependency of the `dist` task. The `test` task's dependencies such as `compileTest` are not executed either. Those dependencies of `test` that are required by another task, such as `compile`, are still executed.
+
+[[sec:rerun_tasks]]
+==== Forcing Tasks To Execute
+
+You can force Gradle to execute all tasks ignoring up-to-date checks using the `--rerun-tasks` option:
+
+----
+❯ gradle test --rerun-tasks
+----
+
+This will force `test` and _all_ task dependencies of `test` to execute. It's a little like running `gradle clean test`, but without the build's generated output being deleted.
+
+[[sec:continue_build_on_failure]]
+==== Continuing the Build When a Failure Occurs
+
+By default, Gradle will abort execution and fail the build as soon as any task fails. This allows the build to complete sooner, but hides other failures that would have occurred. In order to discover as many failures as possible in a single build execution, you can use the `--continue` option.
+
+----
+❯ gradle test --continue
+----
+
+When executed with `--continue`, Gradle will execute _every_ task to be executed where all of the dependencies for that task completed without failure, instead of stopping as soon as the first failure is encountered. Each of the encountered failures will be reported at the end of the build.
+
+If a task fails, any subsequent tasks that were depending on it will not be executed. For example, tests will not run if there is a compilation failure in the code under test; because the test task will depend on the compilation task (either directly or indirectly).
+
+==== Running Tests
+
+Many Gradle plugins like the <<java_plugin, Java plugin>> hook tests to be executed when the `test` task is invoked.
+
+----
+❯ gradle test
+----
+
+If you're using the JUnit and TestNG test frameworks, you can run a subset of tests with the `--tests` option and supplying one or more filters.
+
+----
+❯ gradle test --tests "com.mypackage.foo.*"
+❯ gradle test --tests "*MyTest.myTestMethod"
+❯ gradle someTestTask --tests \*UiTest someOtherTestTask --tests \*WebTest\*ui
+----
+
+Learn more about Java test filtering in the <<test_filtering, Java plugin documentation>>.
+
+This is something you can combine with <<continuous_build, continuous build>> using `--continuous` (or `-t`, for short) to re-execute a subset of tests on every change.
+
+----
+❯ gradle test --continuous --tests "com.mypackage.foo.*"
+----
+
+[[sec:selecting_build]]
+==== Selecting which build to execute
```

Comment:
I may be wrong here but I think the use case is not so common, so doesn't deserve to be mentioned early.

Reply:
Agreed and removed.